lib/std/os.zig

```diff
@@ -359,7 +360,7 @@ pub fn read(fd: fd_t, buf: []u8) ReadError!usize {
             } else {
                 return error.WouldBlock;
             },
-            EBADF => unreachable, // Always a race condition.
+            EBADF => return error.NotOpenForReading, // Can be a race condition.
```

Comment:
I think you should go ahead and apply the same remapping to `EBADF` in WASI as well, since under-the-hood, WASI syscalls will call the same host syscalls and will likely return `EBADF` for the same situation we're handling here.